### PR TITLE
add fileGrp cardinality to ocrd-tool.json schema, deprecate fileGrp names

### PR DESCRIPTION
**Pull request by bertsky**

### CODE REVIEW

**Comment by bertsky** [comment]
@kba I believe I made a mistake in JSON Schema's `oneOf` usage. 

For our simple test JSON …
```JSON
       {                                                                                               
            "git_url": "https://github.com/ocr-d/foo",                                                  
            "version": "0.0.1",                                                                         
            "tools": {                                                                                  
                "ocrd-xyz": {                                                                           
                    "executable": "ocrd-xyz",                                                           
                    "description": "bars all the foos",                                                 
                    "input_file_grp_cardinality": 1,                                                    
                    "output_file_grp_cardinality": 1,                                                   
                    "categories": ["Layout analysis"],                                                  
                    "steps": ["layout/analysis"]                                                        
                }                                                                                       
            }                                                                                           
        }                                  
```
…the CI in core now yields this invalidity:
```
"[tools.ocrd-xyz.input_file_grp_cardinality] 1 is valid under each of {
  'items': {
    'type': 'array', 
    'items': {
      'type': 'number', 
      'multipleOf': 1
    },
    'minItems': 2, 
    'maxItems': 2
  }
}, 
{
  'items': {
    'type': 'number', 'multipleOf': 1
  }
}"
```

So it seems that the list/array syntax got lost...

File: ocrd_tool.schema.yml [hunk 1]
```diff
@@ -29,25 +29,55 @@ properties:
...
+                  type: number
+                  multipleOf: 1
+              - items:
+                  type: array
+                  items:
+                    type: number
+                    multipleOf: 1
+                  minItems: 2
+                  maxItems: 2
+            default: 1
```
Comment (bertsky) [inline ocrd_tool.schema.yml:65]: What's wrong here?

**Review by bertsky — COMMENTED** [review]
pls help

File: ocrd_tool.schema.yml [hunk 2]
```diff
@@ -29,25 +29,51 @@ properties:
...
-          # - output_file_grp
+          - input_file_grp_cardinality
+          - output_file_grp_cardinality
         properties:
           executable:
             description: The name of the CLI executable in $PATH
             type: string
           input_file_grp:
+            deprecated: true
             description: Input fileGrp@USE this tool expects by default
```
Comment (kba) [inline ocrd_tool.schema.yml:40]:
```suggestion
            description: [DEPRECATED] Input fileGrp@USE this tool expects by default
```

see below

File: ocrd_tool.schema.yml [hunk 3]
```diff
@@ -29,25 +29,51 @@ properties:
...
+              - type: number
+                multipleOf: 1
+              - type: array
+                items:
+                  type: number
+                  multipleOf: 1
+                minItems: 2
+                maxItems: 2
+            default: 1
+            additionalProperties: false
```
[the remark this answers is not in the document]
Comment (bertsky) [reply ocrd_tool.schema.yml:64]: Yes, sry, leftover from my internal `object` representation.

**Review by bertsky — COMMENTED** [review]
ah got it – many thanks!

File: ocrd_tool.schema.yml [hunk 4]
```diff
@@ -29,25 +29,51 @@ properties:
...
           input_file_grp:
+            deprecated: true
             description: Input fileGrp@USE this tool expects by default
             type: array
             items:
               type: string
               # pattern: '^OCR-D-[A-Z0-9-]+$'
           output_file_grp:
+            deprecated: true
             description: Output fileGrp@USE this tool produces by default
```
[the remark this answers is not in the document]
Comment (bertsky) [reply ocrd_tool.schema.yml]: wait: that yields
```
yaml.parser.ParserError: while parsing a block mapping
```
culprit are the square brackets

[the remark this answers is not in the document]
Comment (bertsky) [reply ocrd_tool.schema.yml]: fixed via parentheses

**Comment by bertsky** [comment]
I updated this in OCR-D/core#1240 – let's hope CI works now

**Comment by kba** [comment]
GHA Actions tests, pass, I've manually started CircleCI again https://app.circleci.com/pipelines/github/OCR-D/core/2692/workflows/ac341a28-e2e0-45aa-ab00-9d56aa58eb0a Once that passes, I'll merge and release new spec version.

**Pull request merged**